var/spack/repos/builtin/packages/ipopt/package.py

```diff
@@ -9,20 +9,28 @@
 class Ipopt(AutotoolsPackage):
     """Ipopt (Interior Point OPTimizer, pronounced eye-pea-Opt) is a
        software package for large-scale nonlinear optimization."""
-    homepage = "https://projects.coin-or.org/Ipopt"
-    url      = "http://www.coin-or.org/download/source/Ipopt/Ipopt-3.12.4.tgz"
 
+    homepage = "https://github.com/coin-or/Ipopt"
+    url      = "https://www.coin-or.org/download/source/Ipopt/Ipopt-3.13.2.tgz"
+    #Alternative: url      = "https://github.com/coin-or/Ipopt/archive/releases/3.13.2.tar.gz"
```

Comment:
You can use:
```python
urls = [
    "https://github.com/coin-or/Ipopt/archive/releases/3.13.2.tar.gz",
    "https://www.coin-or.org/download/source/Ipopt/Ipopt-3.13.2.tgz",
]
```
and assuming they have the same checksums, Spack will try the first one, and if there are any download issues try the second one.

Reply:
Precisely, they don’t all have the same checksum.

Reply:
Let me paste here the last commit message:
"Pagination on Github prevent spack from easily parse all available
versions. Also, due to recent migration to GitHub, tarballs for
versions up to 3.12.13 have be regenerated, changing the hash.

The current URL will apparently be supported, so we keep it, and give
the alternative one as a comment."

Reply:
That’s not really beautiful, I confess.